tsdb/head.go

```diff
@@ -586,32 +591,38 @@ func (h *Head) Init(minValidTime int64) error {
 	}
 	// Backfill segments from the most recent checkpoint onwards.
 	for i := startFrom; i <= endAt; i++ {
```

Comment:
We can keep the larger piece with less indentation (and easier to read diff) by having this as
```go
	// Backfill segments from the most recent checkpoint onwards.
	for i := startFrom; i <= endAt; i++ {
		select {
		case <-ctx.Done():
			level.Info(h.logger).Log("msg", "Aborting WAL replay")
			return ErrWALReplayAborted
		default:
		}
		// ... everything else
	}
```

Reply:
That's done